### ISSUE TITLE
Parallelize PointSetToPointSetMetricv4

### ISSUE THREAD

**Issue by dzenanz** [InsightSoftwareConsortium/ITK#1144]
This was already done by 687c49176ba97fbbe805fc468bd96ec36b46ffc0, but had to be reverted by #1139 because #1138 did not fix sporadic failures #1131.

**Comment by dzenanz** [comment]
I created this issue so we don't forget about it. @thewtex @hjmjohnson @samuelgerber

**Comment by samuelgerber** [comment]
So I don't forget what I tried to fix it:
1. derivative size is wrong at https://github.com/InsightSoftwareConsortium/ITK/commit/687c49176ba97fbbe805fc468bd96ec36b46ffc0#r34604258
but this was not the issue.
2. It appeared that running it with only one thread fixed the issue (need to double check). Some concurrency issue somewhere?
3. Tested if m_VirtualTransformedPointSet is updated concurrently and thread has wrong address to invalid array. I.e. the update checks might be incorrect when resetting the point sets etc. of the metric.
4. Tried to store derivative field individually per thread and sum after into derivative.
